vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -276,21 +337,30 @@ func matchNameforVMDK(mount_source string) bool {
 	return false
 }
 
-// enumberates volumes and  builds RefCountsMap, then sync with mount info
-func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver) error {
+// check if refcounting has been made dirty by mounts/unmounts
+func (r *RefCountsMap) checkDirty() bool {
+	r.StateMtx.Lock()
+	defer r.StateMtx.Unlock()
+	return r.isDirty
+}
+
+// enumerates volumes and  builds RefCountsMap, then sync with mount info
+func (r *RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver) error {
 	// we assume to  have empty refcounts. Let's enforce
 
-	r.mtx.Lock() // Lock the RefCountsMap to purge the refcounts
-	for name := range r.refMap {
-		delete(r.refMap, name)
-	}
-	r.mtx.Unlock() // Unlock.
+	r.StateMtx.Lock()
+	r.isDirty = false
+	r.StateMtx.Unlock()
 
 	filters := filters.NewArgs()
 	filters.Add("status", "running")
 	filters.Add("status", "paused")
 	filters.Add("status", "restarting")
 
+	if r.checkDirty() {
```

Comment:
Creating filter is lightweight operation. No need to check isDirty here. You can remove this check.